client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -598,11 +607,13 @@ function getProfessionalEmailUpsellUrl( {
 	}
 
 	if (
+		! isDomainOnly &&
 		! hasBloggerPlan( cart ) &&
 		! hasPersonalPlan( cart ) &&
 		! hasBusinessPlan( cart ) &&
 		! hasEcommercePlan( cart ) &&
-		! hasProPlan( cart )
+		! hasProPlan( cart ) &&
+		! hasStarterPlan( cart )
```

Comment:
The addition of `hasStarterPlan` was not related directly to this PR, but I didn't see any reason why we should have the Starter plan as part of this condition, so I added it anyway.